dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileTypeDetector.java

```diff
@@ -199,4 +207,10 @@ private boolean isGraphMLFile(File file) {
         logger.debug("end isGraphML()");
         return isGraphML;
     }
+
+    // For tests
+    FileTypeDetector withSettingsService(SettingsServiceBean settingsService) {
+        this.settingsService = settingsService;
+        return this;
+    }
```

Comment:
I used this approach specifically because of the standard constructor requirement. Seemed cleaner for the "normal" production use-case. But now that I think about it, the convention with constructor might be clearer. I'm changing it.